src/js/helpers/useToast.ts

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ */
+
+import { Toast } from 'bootstrap';
+import Toaster from '@constants/useToast-data';
+import selectorsMap from '@constants/selectors-map';
+
+const useToast = (message: string, options?: Toaster.Option): Toast => {
+  const toastElement = getToastElement();
+
+  addToastMessage(toastElement, message);
+
+  options = {...Toaster.Default, ...options};
+
+  addToastClassList(toastElement, options);
+
+  if (options.autohide === false) {
+    showToastBtnClose(toastElement);
+  }
+ 
+  return new Toast(toastElement, { autohide: options.autohide, delay: options.delay });
+}
+
+const getToastElement = (): HTMLElement => {
+  const toastContainer = document.querySelector<HTMLElement>(selectorsMap.toast.container);
+
+  if (toastContainer) {
+    const toastTemplate = toastContainer.querySelector<HTMLTemplateElement>(selectorsMap.toast.template);
+    const toastClone = toastTemplate?.content.cloneNode(true) as DocumentFragment;
+    const toastElement = toastClone?.querySelector<HTMLElement>(selectorsMap.toast.toast);
+    const toastBody = toastElement?.querySelector<HTMLElement>(selectorsMap.toast.body);
+
+    if (toastElement && toastBody) {
+      toastContainer.appendChild(toastElement);
+  
+      return toastElement;
+    } else {
+      toastContainer.remove();
```

Comment:
```suggestion
      toastContainer.remove();
```

Don't you need to verify if it exist before removing it?

Reply:
Line 49: will be verified.

Reply:
My bad, but then there is another problem, what happen if the container is undefined? It's not generated right?

There are some edge cases where child theme can still remove the container, it should be also added on the fly if it doesn't exist!

It will be done only one time, as the first toast use will inject it, then next toasts won't need to inject it

Reply:
When fallback will be used:

1. toast-container is not exists
2. toast-container is exists but toast template has invalid structure

Reply:
Oh because you remove the container

I don't really like it, imagine that the container has some new classes added by the child theme for any reason?

I would verify if the container exist and I would generate it only if it doesn't exist

Reply:
Why someone need to add custom class to toast container, the position is fixed, it doesn't have any text content, or style rules,  just will be used by BS5 Toast for finding toasts.

Reply:
If they want the toast system to be on the bottom right for example, instead of top right :)

Reply:
You'll end by just erasing it, because it misses the toast template in some way

I agree it's an edge case, but it can be handled so..

Reply:
Oh man!
You right.
Let me think about that.

Reply:
If `toast-container` exists, but `toast-template` is not exist or is not valid.,
then remove inner html of `toast-container` and replace with inner html of fallback-container.

Wdyt?

Reply:
You'll have some problem if someone or a module did use `useToast` before you, because it will erase their toast instance, meaning that when they'll throw `myToast.show()`, it won't be shown anymore because the inject markup is not in the `.toast-container` anymore

Reply:
I think, if `toast-container` exists, just happen the toast markup to it, if it doesn't exist, append a `toast-container` to the body and then append the `toast-template` content to it!

Reply:
You should probably separate it to 2 different logic:
- The toast-container check (and append to the body if it doesn't exist)
- The toast template check and append it to the toast container

Reply:
useToast is not static (`new Toast()`), for each request an instance will be created, before creating a toast, this modification will be done.

Reply:
You'll end up with a function: `useToastContainer` for example, which will handle all the check and generation then return the `toast-container` element so you can use it

And another one: `useToast`, which will use `useToastContainer` on the top

Reply:
instead of `useToastContainer`

What about a generic `useContainer` function? Taking a selector as a parameter, and a template content (for the fallback), and an `appendTo` selector? It can be used both for `useToast` and `useAlert`

Reply:
I love generics :smile: 
I'll work on it.